pandas/core/indexing.py

```diff
@@ -1898,16 +1898,20 @@ def _setitem_with_indexer_2d_value(self, indexer, value):
 
         ilocs = self._ensure_iterable_column_indexer(indexer[1])
 
-        # GH#7551 Note that this coerces the dtype if we are mixed
```

Comment:
is the GH reference here worth keeping somewhere?

Reply:
Don't think so, since we are avoiding this now